rust/src/io/reader.rs

```diff
@@ -234,7 +235,23 @@ impl<'a> FileReader<'a> {
         let offset_arr = subtract_scalar(positions, start_position)?;
         let value_arrs = self.read_array(&field.children[0], batch_id).await?;
 
-        Ok(Arc::new(ListArray::new(value_arrs, &offset_arr)?))
+        Ok(Arc::new(ListArray::try_new(value_arrs, &offset_arr)?))
+    }
+
+    // TODO: merge with [read_list_array]?
```

Comment:
could potentially create a read_generic_list_array<T> that both methods call with Int32 or Int64?

Reply:
Yes, trying to use int32/int64 as template parameter to see how to get it working.